base/special/exp.jl

```diff
@@ -198,7 +198,11 @@ end
 # The solution is to do the scaling back in 2 steps as just messing with the exponent wouldn't work.
 for (func, base) in (:exp2=>Val(2), :exp=>Val(:ℯ), :exp10=>Val(10))
     @eval begin
-        ($func)(x::Real) = ($func)(float(x))
+        function ($func)(x::Real)
+            xf = float(x)
+            x === xf && throw(MethodError($func, (x,)))
+            return ($func)(xf)
+        end
```

Comment:
why not add this to the other list?

Reply:
It breaks `mathconstants.jl`

Reply:
How so?

Reply:
The constant there that relies on `exp(big())`